org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WTableAttribute.java

```diff
@@ -0,0 +1,387 @@
+		grid.makeNoStrip();
+		grid.setOddRowSclass("even");
+		vbox.appendChild(grid);
+
+		rows = new Rows();
+		grid.appendChild(rows);
+
+		// ConfirmPanel
+		confirmPanel.addActionListener(this);
+		vbox.appendChild(confirmPanel);
+	} // jbInit
+
+	/**
+	 * Dynamic Init of the Center Panel
+	 */
+	private void dynInit()
+	{
+		addAttributes();
+	} // dynInit
+
+	/**
+	 * Add Attributes
+	 * 
+	 * @return rows
+	 */
+	private int addAttributes()
+	{
+		PO po = MTable.get(Env.getCtx(), p_AD_Table_ID).getPO(p_Record_ID, null);
+		MAttribute[] attributes = MAttributeSet.get(Env.getCtx(), p_M_AttributeSet_ID).getMAttributes(false);
+		for (MAttribute attribute : attributes)
+		{
+
+			WEditor editor = null;
+			//
+			if (MAttribute.ATTRIBUTEVALUETYPE_List.equals(attribute.getAttributeValueType()))
+			{
+				editor = WebEditorFactory.getEditor(getListTypeGridField(attribute), true);
+			}
+			else if (MAttribute.ATTRIBUTEVALUETYPE_Number.equals(attribute.getAttributeValueType()))
+			{
+				editor = WebEditorFactory.getEditor(getNumberGridField(attribute), true);
+			}
+			else if (MAttribute.ATTRIBUTEVALUETYPE_Reference.equals(attribute.getAttributeValueType()))
+			{
+				editor = WebEditorFactory.getEditor(getGridField(attribute), true);
+			}
+			else if (MAttribute.ATTRIBUTEVALUETYPE_Date.equals(attribute.getAttributeValueType()))
+			{
+				editor = WebEditorFactory.getEditor(getDateGridField(attribute), true);
+			}
+			else // Text Field
+			{
+				editor = WebEditorFactory.getEditor(getStringGridField(attribute), true);
+			}
+
+			if (editor != null)
+			{
+				Row row = rows.newRow();
+
+				Label label = editor.getLabel();
+				if (!Util.isEmpty(attribute.getDescription(), true))
+					label.setValue(attribute.getDescription());
+				else if (Util.isEmpty(label.getValue(), true))
+					label.setValue(attribute.getName());
+
+				row.appendCellChild(label.rightAlign());
+
+				editor.setMandatory(attribute.isMandatory());
+				editor.fillHorizontal();
+				editor.setValue(po.getAttribute(attribute.getName()));
+				editor.addValueChangeListener(new ValueChangeListener() {
+
+					@Override
+					public void valueChange(ValueChangeEvent evt)
+					{
+						if (evt.getSource() instanceof WEditor)
+						{
+							WEditor sourceEditor = (WEditor) evt.getSource();
+							// IDEMPIERE-2999 - set value in online button as HRef
+							if (sourceEditor.getGridField().getDisplayType() == DisplayType.URL)
+								((Urlbox) sourceEditor.getComponent()).setText((String) evt.getNewValue());
+						}
+					}
+				});
+
+				Component fieldEditor = editor.getComponent();
+				row.appendCellChild(fieldEditor, 2);
+				editor.showMenu();
+				m_attEditors.put(attribute.getM_Attribute_ID(), editor);
+			}
+		}
+		return 0;
+	} // addProductAttributes
+
+	/**
+	 * Get Field editor for attribute
+	 * @param attribute
+	 * @return
+	 */
+	public GridField getGridField(MAttribute attribute)
+	{
+		GridFieldVO vo = GridFieldVO
```

Comment:
for some reason the Msg.translate here is not working well and the label is not translated.

Maybe this patch is required:
```
diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WTableAttribute.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WTableAttribute.java
index 980ef9e079..83b8bff562 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WTableAttribute.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WTableAttribute.java
@@ -184,7 +184,7 @@ public class WTableAttribute extends Window implements EventListener<Event>
 				if (!Util.isEmpty(attribute.getDescription(), true))
 					label.setValue(attribute.getDescription());
 				else if (Util.isEmpty(label.getValue(), true))
-					label.setValue(attribute.getName());
+					label.setValue(Msg.translate(Env.getCtx(), attribute.get_Translation("Name")));
 
 				row.appendCellChild(label.rightAlign());
 
```